instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/snippet/InjectionObject.java

```diff
@@ -0,0 +1,73 @@
+package io.opentelemetry.javaagent.instrumentation.servlet.v3_0.snippet;
+
+import io.opentelemetry.javaagent.bootstrap.servlet.SnippetHolder;
+import java.io.IOException;
+import javax.servlet.ServletOutputStream;
+
+public class InjectionObject {
+  public int headTagBytesSeen = -1;
+  public String characterEncoding;
+  public SnippetInjectingResponseWrapper wrapper;
+
+  public boolean inject() {
+    return headTagBytesSeen == 5;
+  }
+
+  public boolean injected() {
+    return headTagBytesSeen == -2;
+  }
+
+  public void intInjectionHelper(byte b) {
+    int headTagBytesSeen = this.headTagBytesSeen;
+    if (headTagBytesSeen == -1 && b == '<') {
+      headTagBytesSeen = 0;
+    } else if (headTagBytesSeen == 0 && b == 'h') {
+      headTagBytesSeen = 1;
+    } else if (headTagBytesSeen == 1 && b == 'e') {
+      headTagBytesSeen = 2;
+    } else if (headTagBytesSeen == 2 && b == 'a') {
+      headTagBytesSeen = 3;
+    } else if (headTagBytesSeen == 3 && b == 'd') {
+      headTagBytesSeen = 4;
+    } else if (headTagBytesSeen == 4 && b == '>') {
+      headTagBytesSeen = 5;
+    } else if (b > 0 && headTagBytesSeen != -2) {
+      headTagBytesSeen = -1;
+    }
+    this.headTagBytesSeen = headTagBytesSeen;
+  }
+
+  public boolean intInjection(ServletOutputStream sp, byte b) throws IOException {
+    intInjectionHelper(b);
+    if (inject()) {
+      headTagBytesSeen = -2;
```

Comment:
(or ALREADY_INJECTED_FAKE_VALUE)
```suggestion
      headTagBytesSeen = ALREADY_INJECTED;
```